pilot/platform/kube/controller.go

```diff
@@ -359,6 +383,11 @@ func (c *Controller) HostInstances(addrs map[string]bool) ([]*model.ServiceInsta
 						if exists {
 							az, _ = c.GetPodAZ(pod)
 							sa = kubeToIstioServiceAccount(pod.Spec.ServiceAccountName, pod.GetNamespace(), c.domainSuffix)
+							glog.V(2).Infof("podname (%s, %s), namespace(%s, %s)", podname, pod.GetName(), namespace, pod.GetNamespace())
+							if podname != pod.GetName() || namespace != pod.GetNamespace() {
```

Comment:
I think there might be a race here too. Keep in mind, that read/writes to kube API server are asynchronous. We might not have the fresh data, even if some other pod has made the write.